### ISSUE TITLE
Pulses not loading anymore

### ISSUE THREAD

**Issue by DieterMartin** [metabase/metabase#9891]
------
Before filing an issue we'd appreciate it if you could take a moment to ensure
there isn't already an open issue or pull-request.
-----

If there's an existing issue, please add a :+1: reaction to the description of
the issue. One way we prioritize issues is by the number of :+1: reactions on
their descriptions. Please DO NOT add `+1` or :+1: comments.

### Security Disclosure

Security is very important to us. If discover any issue regarding security, please disclose the information responsibly by sending an email to security@metabase.com and not by creating a GitHub issue.

### Database support

For requests for new database drivers, please include the name
of the database and the version. Additionally, giving us a sense of how and what
you use it for, and other high-level details about the database, can help us get
a better picture of what would need to done to support it.

### Feature requests and proposals
We're excited to hear how we can make Metabase better. Please add as much detail
as you can on your use case.

### Bugs
If you're filing an issue about a bug please include as much information
as you can including the following.

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, IE 10)
- Your operating system: (e.x. OS X 10, Windows XP, etc)
- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
- Metabase version: (e.x. 0.19.3)
- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Linux/Ubuntu 12)
- Metabase internal database: (e.x. H2 (default), MySQL, Postgres)

- *Repeatable steps to reproduce the issue*

Thanks for being part of the Metabase project!
-------

**Comment by DieterMartin** [comment]
Your browser and the version: Chrome 73.0.3683.103

Your operating system:  OS X 10.14.4

Your databases:  Postgres on AWS RDS

Metabase version: v0.32.5.metabase-aws-eb.zip

Metabase hosting environment: Elastic Beanstalk

Metabase internal database: H2 i believe

Repeatable steps to reproduce the issue:

i have recently set up metabase and worked great.  set up several dashboards and questions with no problems.  set up pulses which worked great.  test messages went through.  the following morning (no other changes made) the pulses did not work.  loading the pulses to look at them, the question does not load with spinning loop as:
<img width="358" alt="Screen Shot 2019-05-05 at 12 19 28 PM" src="https://user-images.githubusercontent.com/24588204/57197696-13094a00-6f30-11e9-838a-78dae130e809.png">

the question still works great when looked at directly.  this seems very similar to closed issue #5799.  looking at the java dev tools: 
![Screen Shot 2019-05-05 at 12 21 28 PM](https://user-images.githubusercontent.com/24588204/57197734-74311d80-6f30-11e9-95be-9a768d143957.png)

deleting the pulse and starting over doesn't resolve issue.   pulses globally not working.  i am not using slack, am using email pulses only.  test email in admin screen works with no problem.  i looked through logs in admin/troubleshooting and nothing stood out (but i'm not sure what i would be looking for)

happy to give more information if needed.  any advice for workaround is def appreciated.

**Comment by flamber** [comment]
Hi @DieterMartin 
The javascript errors looks to come from various http responses, so the full errors should be available if you look at Admin Settings > Troubleshooting > Log. You should be looking for anything in the log that is around the time of the problem and with the type `WARN` or `ERROR`.
You can also look at the logs directly in `/var/log/metabase` or via your [EB control panel](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/using-features.logging.html)
Some people have reported that killing the database connection fixes the problem (temporarily). Can also be done by restarting Metabase.
Please be careful with using H2 on EB - a lot of people have lost their data because they didn't have backups, when upgrading Metabase, so either place the H2 on a different volume or [migrate to a different](https://www.metabase.com/docs/latest/operations-guide/start.html#migrating-from-using-the-h2-database-to-mysql-or-postgres) backend database.

**Comment by DieterMartin** [comment]
thank you for your response.   i have restarted metabase to no avail.  by killing database connection do you mean the database that i am sampling from or metabase's database.  the metabase db is on aws rds, it is postgres so i was mistaken on the H2 statement.

looking through the logs, the following represents a majority of the "WARN" entries:

May 05 16:58:38 WARN metabase.query-processor.middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.
java.lang.Exception: Invalid response from database driver. No :status provided.true
at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)
at clojure.core$partial$fn__5828.invoke(core.clj:2638)
at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)
at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)
at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)
at clojure.lang.AFn.run(AFn.java:22)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
May 05 16:58:38 DEBUG metabase.middleware.log :: POST /api/dataset 200 [ASYNC: completed] 139 ms (9 DB calls) Jetty threads: 8/50 (2 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:38 DEBUG metabase.async.api-response :: Async response finished, closing channels.
May 05 16:58:40 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:40 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:40 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:40 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 7 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 7 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 7 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 7 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 6 idle, 0 queued) (49 total active threads)
May 05 16:58:41 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 6 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 6 idle, 0 queued) (48 total active threads)
May 05 16:58:42 DEBUG metabase.middleware.log :: GET /api/database/2/autocomplete_suggestions 200 7 ms (5 DB calls) Jetty threads: 8/50 (3 busy, 6 idle, 0 queued) (48 total active threads)
May 05 16:58:42 DEBUG metabase.async.semaphore-channel :: Permit available without waiting, will run fn immediately
May 05 16:58:42 DEBUG metabase.async.util :: Running metabase.query_processor.middleware.async$async__GT_sync$fn__33067@11410a71 on separate thread...
May 05 16:58:42 DEBUG metabase.async.api-response :: starting streaming response
May 05 16:58:42 ERROR metabase.driver.sql-jdbc.execute :: Failed to set timezone: 
SQLException:
Message: (conn=1843) Unknown or incorrect time zone: 'US/Central'
SQLState: HY000
Error Code: 1298

thanks again.  let me know if you need more information.

**Comment by flamber** [comment]
Restarting Metabase will also restart all database connections, so if that didn't help, then killing individual connections probably won't make any difference.

The error `Unhandled exception, exepected catch-exceptions middleware to handle it` is a known error and will likely get fixed in 0.32.6 (#9818) - I'm pretty sure you can ignore that.

The rest of the log you posted, doesn't seem to relate to the javascript error you saw in the dev console.
You need to produce the error again (by going to your pulse) and then check the log.

**Comment by DieterMartin** [comment]
going to the pulse does not generate anything in the log.  the error in the java console complains that:

Refused to load manifest from 'https://metabase.metacinegroup.com/app/assets/img/site.webmanifest' because it violates the following Content Security Policy directive: "default-src 'none'". Note that 'manifest-src' was not explicitly set, so 'default-src' is used as a fallback.

looking at https://metabase.metacinegroup.com/app/assets/img/site.webmanifest, it contains:

########
{
    "name": "Metabase",
    "short_name": "Metabase",
    "icons": [
        {
            "src": "/app/assets/img/android-chrome-192x192.png",
            "sizes": "192x192",
            "type": "image/png"
        },
        {
            "src": "/app/assets/img/android-chrome-256x256.png",
            "sizes": "256x256",
            "type": "image/png"
        }
    ],
    "theme_color": "#ffffff",
    "background_color": "#ffffff",
    "display": "standalone"
}
so perhaps there needs to be some statement of 'manifest-src' in that file?  it also notes that default-src is set to none, is that a setting that needs to be declared somewhere?  

again, loading any pulse hangs as described and there is no note of the event in the logs for metabase

**Comment by DieterMartin** [comment]
correcting my settings: (my db is mysql, metabase is postgres)

Your browser and the version: Chrome 73.0.3683.103

Your operating system: OS X 10.14.4

Your databases: mysql

Metabase version: v0.32.5.metabase-aws-eb.zip

Metabase hosting environment: Elastic Beanstalk

Metabase internal database: postgres on AWS RDS

**Comment by flamber** [comment]
You can completely ignore the error "Refused to load manifest" - that is only related to Android.
And your browser console shows JavaScript notifications - not Java - sounds similar, but totally different.

So since you're not seeing any errors anymore, does that mean the restart actually worked?
I was interested in the actual error, that the server would have created, when the browser received the errors "Failed to load resource: the server responded with a status 50x"

**Comment by DieterMartin** [comment]
got it.  i reloaded browser at 22:35 on this log, giving a little runnup to the event:

May 05 22:34:42 DEBUG metabase.middleware.log :: GET /api/session/properties 200 3 ms (1 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (46 total active threads)
May 05 22:34:42 DEBUG metabase.middleware.log :: GET /api/user/current 200 5 ms (3 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (46 total active threads)
May 05 22:34:42 DEBUG metabase.middleware.log :: GET /api/user/current 200 5 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (46 total active threads)
May 05 22:34:42 DEBUG metabase.middleware.log :: GET /api/database 200 6 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (46 total active threads)
May 05 22:34:43 DEBUG metabase.middleware.log :: GET /api/setting 200 496 µs (0 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (46 total active threads)
May 05 22:35:54 DEBUG metabase.middleware.log :: GET /api/session/properties 200 3 ms (1 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (44 total active threads)
May 05 22:35:54 DEBUG metabase.middleware.log :: GET /api/user/current 200 7 ms (3 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/user/current 200 5 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/database 200 5 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/user 200 14 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/setting 200 420 µs (0 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/pulse/form_input 200 349 µs (0 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/pulse/1 200 13 ms (7 DB calls) Jetty threads: 8/50 (3 busy, 3 idle, 0 queued) (44 total active threads)
May 05 22:35:55 DEBUG metabase.async.semaphore-channel :: Permit available without waiting, will run fn immediately
May 05 22:35:55 DEBUG metabase.async.util :: Running metabase.query_processor.middleware.async$async__GT_sync$fn__33067@764df387 on separate thread...
May 05 22:35:55 DEBUG metabase.middleware.log :: GET /api/collection/1 200 10 ms (3 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (44 total active threads)
May 05 22:35:55 ERROR metabase.driver.sql-jdbc.execute :: Failed to set timezone: 
SQLException:
Message: (conn=1960) Unknown or incorrect time zone: 'US/Central'
SQLState: HY000
Error Code: 1298

May 05 22:35:55 WARN metabase.query-processor.middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.
java.lang.Exception: Invalid response from database driver. No :status provided.true
at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)
at clojure.core$partial$fn__5828.invoke(core.clj:2638)
at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)
at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)
at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)
at clojure.lang.AFn.run(AFn.java:22)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
May 05 22:35:55 ERROR metabase.middleware.log :: GET /api/pulse/preview_card_info/3 500 296 ms (10 DB calls) Jetty threads: 8/50 (3 busy, 3 idle, 0 queued) (44 total active threads) 
{:message "TypeError: formatter is not a function",
:type jdk.nashorn.internal.runtime.ECMAException,
:stacktrace
("jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)"
"jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:213)"
"jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:185)"
"jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:172)"
"jdk.nashorn.internal.runtime.Undefined.lookup(Undefined.java:102)"
"jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(NashornLinker.java:106)"
"jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(NashornLinker.java:98)"
"jdk.internal.dynalink.support.CompositeTypeBasedGuardingDynamicLinker.getGuardedInvocation(CompositeTypeBasedGuardingDynamicLinker.java:176)"
"jdk.internal.dynalink.support.CompositeGuardingDynamicLinker.getGuardedInvocation(CompositeGuardingDynamicLinker.java:124)"
"jdk.internal.dynalink.support.LinkerServicesImpl.getGuardedInvocation(LinkerServicesImpl.java:154)"
"jdk.internal.dynalink.DynamicLinker.relink(DynamicLinker.java:253)"
"jdk.nashorn.internal.scripts.Script$Recompilation$343$131978AAA$\\^eval\\_.L:10#L:4262#makeCellBackgroundGetter#L:4301(<eval>:4306)"
"jdk.nashorn.internal.scripts.Script$Recompilation$342$129273AAA$\\^eval\\_.L:10#L:4187#L:4190#makeCellBackgroundGetter#L:4210(<eval>:4211)"
"jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:643)"
"jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494)"
"jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393)"
"jdk.nashorn.api.scripting.ScriptObjectMirror.call(ScriptObjectMirror.java:117)"
"--> pulse.color$get_background_color.invokeStatic(color.clj:60)"
"pulse.color$get_background_color.invoke(color.clj:56)"
"pulse.render$render_table$fn__26610$fn__26612.invoke(render.clj:373)"
"api.pulse$fn__58362.invokeStatic(pulse.clj:162)"
"api.pulse$fn__58362.invoke(pulse.clj:155)"
"middleware.auth$enforce_authentication$fn__60228.invoke(auth.clj:14)"
"routes$fn__61825$fn__61826.doInvoke(routes.clj:113)"
"middleware.exceptions$catch_uncaught_exceptions$fn__60319.invoke(exceptions.clj:104)"
"middleware.exceptions$catch_api_exceptions$fn__60316.invoke(exceptions.clj:92)"
"middleware.log$log_api_call$fn__61698$fn__61699.invoke(log.clj:168)"
"middleware.log$log_api_call$fn__61698.invoke(log.clj:162)"
"middleware.security$add_security_headers$fn__60282.invoke(security.clj:102)"
"middleware.json$wrap_json_body$fn__61403.invoke(json.clj:61)"
"middleware.json$wrap_streamed_json_response$fn__61421.invoke(json.clj:97)"
"middleware.session$bind_current_user$fn__58938$fn__58939.invoke(session.clj:188)"
"middleware.session$do_with_current_user.invokeStatic(session.clj:171)"
"middleware.session$do_with_current_user.invoke(session.clj:165)"
"middleware.session$bind_current_user$fn__58938.invoke(session.clj:187)"
"middleware.session$wrap_current_user_id$fn__58927.invoke(session.clj:156)"
"middleware.session$wrap_session_id$fn__58912.invoke(session.clj:118)"
"middleware.auth$wrap_api_key$fn__60236.invoke(auth.clj:27)"
"middleware.misc$maybe_set_site_url$fn__61761.invoke(misc.clj:55)"
"middleware.misc$bind_user_locale$fn__61764.invoke(misc.clj:71)"
"middleware.misc$add_content_type$fn__61751.invoke(misc.clj:30)"
"middleware.misc$disable_streaming_buffering$fn__61786.invoke(misc.clj:116)"
"middleware.misc$wrap_gzip$fn__61779.invoke(misc.clj:98)")}

May 05 22:35:58 DEBUG metabase.middleware.log :: GET /api/session/properties 200 49 ms (1 DB calls) Jetty threads: 8/50 (4 busy, 2 idle, 0 queued) (44 total active threads)
May 05 22:35:58 DEBUG metabase.middleware.log :: GET /api/user/current 200 52 ms (3 DB calls) Jetty threads: 8/50 (4 busy, 2 idle, 0 queued) (44 total active threads)
May 05 22:35:59 DEBUG metabase.middleware.log :: GET /api/user/current 200 5 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)
May 05 22:35:59 DEBUG metabase.middleware.log :: GET /api/database 200 5 ms (3 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)
May 05 22:35:59 DEBUG metabase.middleware.log :: GET /api/setting 200 431 µs (0 DB calls) Jetty threads: 8/50 (3 busy, 4 idle, 0 queued) (44 total active threads)

**Comment by flamber** [comment]
Perfect. Haven't seen that error before. The interesting part is this:
```
May 05 22:35:55 ERROR metabase.middleware.log :: GET /api/pulse/preview_card_info/3 500 296 ms (10 DB calls) Jetty threads: 8/50 (3 busy, 3 idle, 0 queued) (44 total active threads)
{:message "TypeError: formatter is not a function",
:type jdk.nashorn.internal.runtime.ECMAException,
:stacktrace
("jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)"
"jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:213)"
"jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:185)"
"jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:172)"
"jdk.nashorn.internal.runtime.Undefined.lookup(Undefined.java:102)"
"jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(NashornLinker.java:106)"
"jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(NashornLinker.java:98)"
"jdk.internal.dynalink.support.CompositeTypeBasedGuardingDynamicLinker.getGuardedInvocation(CompositeTypeBasedGuardingDynamicLinker.java:176)"
"jdk.internal.dynalink.support.CompositeGuardingDynamicLinker.getGuardedInvocation(CompositeGuardingDynamicLinker.java:124)"
"jdk.internal.dynalink.support.LinkerServicesImpl.getGuardedInvocation(LinkerServicesImpl.java:154)"
"jdk.internal.dynalink.DynamicLinker.relink(DynamicLinker.java:253)"
"jdk.nashorn.internal.scripts.Script$Recompilation$343$131978AAA$\^eval\.L:10#L:4262#makeCellBackgroundGetter#L:4301(:4306)"
"jdk.nashorn.internal.scripts.Script$Recompilation$342$129273AAA$\^eval\.L:10#L:4187#L:4190#makeCellBackgroundGetter#L:4210(:4211)"
"jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:643)"
"jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494)"
"jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393)"
"jdk.nashorn.api.scripting.ScriptObjectMirror.call(ScriptObjectMirror.java:117)"
"--> pulse.color$get_background_color.invokeStatic(color.clj:60)"
"pulse.color$get_background_color.invoke(color.clj:56)"
"pulse.render$render_table$fn__26610$fn__26612.invoke(render.clj:373)"
"api.pulse$fn__58362.invokeStatic(pulse.clj:162)"
"api.pulse$fn__58362.invoke(pulse.clj:155)"
"middleware.auth$enforce_authentication$fn__60228.invoke(auth.clj:14)"
"routes$fn__61825$fn__61826.doInvoke(routes.clj:113)"
"middleware.exceptions$catch_uncaught_exceptions$fn__60319.invoke(exceptions.clj:104)"
"middleware.exceptions$catch_api_exceptions$fn__60316.invoke(exceptions.clj:92)"
"middleware.log$log_api_call$fn__61698$fn__61699.invoke(log.clj:168)"
"middleware.log$log_api_call$fn__61698.invoke(log.clj:162)"
"middleware.security$add_security_headers$fn__60282.invoke(security.clj:102)"
"middleware.json$wrap_json_body$fn__61403.invoke(json.clj:61)"
"middleware.json$wrap_streamed_json_response$fn__61421.invoke(json.clj:97)"
"middleware.session$bind_current_user$fn__58938$fn__58939.invoke(session.clj:188)"
"middleware.session$do_with_current_user.invokeStatic(session.clj:171)"
"middleware.session$do_with_current_user.invoke(session.clj:165)"
"middleware.session$bind_current_user$fn__58938.invoke(session.clj:187)"
"middleware.session$wrap_current_user_id$fn__58927.invoke(session.clj:156)"
"middleware.session$wrap_session_id$fn__58912.invoke(session.clj:118)"
"middleware.auth$wrap_api_key$fn__60236.invoke(auth.clj:27)"
"middleware.misc$maybe_set_site_url$fn__61761.invoke(misc.clj:55)"
"middleware.misc$bind_user_locale$fn__61764.invoke(misc.clj:71)"
"middleware.misc$add_content_type$fn__61751.invoke(misc.clj:30)"
"middleware.misc$disable_streaming_buffering$fn__61786.invoke(misc.clj:116)"
"middleware.misc$wrap_gzip$fn__61779.invoke(misc.clj:98)")}
```

**Comment by camsaul** [comment]
@tlrobinson do you know anything about this error on 0.32.x? 

```
{:message "TypeError: formatter is not a function",
:type jdk.nashorn.internal.runtime.ECMAException,
```

it's in the code that renders Pulses

```
pulse.color$get_background_color.invokeStatic(color.clj:60)
```

**Comment by linhnn** [comment]
Hi, we have the same issue like that. @tlrobinson can you help to check ? The error from the function **get-background-color**
`
src/metabase/pulse/color.clj
`

**Comment by camsaul** [comment]
@tlrobinson bump in case you didn't see my last message

**Comment by flamber** [comment]
From the [forum](https://discourse.metabase.com/t/pulse-to-slack-doesnt-show-data/6705). This was triggered after field type was changed from `Description` to `No special type`, since pulses doesn't include `Description` and that field was needed in the pulse. Changing it back to `Description` fixed the error.

**Comment by tlrobinson** [comment]
I haven't been able to reproduce this by creating a field that's marked as `Description` then changing to a different `special_type` (or none). Does anyone have specific reproduction steps?

**Comment by walterl** [comment]
I think I found the issue: `resources/frontend_shared/color_selector.js` is outdated.

The current version does not support all the formatting operators that are defined in `frontend/src/metabase/visualizations/lib/table_format.js`. Since the outdated file is only used by the back-end, it explains why the pulses only fail when rendered by the back-end, and not when the test pulses are rendered in the front-end.

So, to reproduce the issue:
* Apply any of the following operators to a pulse question: `is-null`, `not-null`, `contains`, `does-not-contain`, `starts-with`, `ends-with`. (I didn't test all of them.)
* Trigger a pulse render on the back-end. I used the following code in a repl:

```clojure
(ns user
  (:require [metabase.email.messages :as messages]
            [metabase.models.pulse :refer [retrieve-notification Pulse]]
            [metabase.pulse :as p]))

(let [pulse    (retrieve-notification 877 :archived false)
      results  (for [card  (:cards pulse)
                     :let  [result (p/execute-card (:id card), :pulse-id (:id pulse))]
                     :when result]
                 result)
      timezone (-> results first :card p/defaulted-timezone)]
  (messages/render-pulse-email timezone pulse results))
```

Running `yarn build-shared` to update `resources/frontend_shared/color_selector.js` fixes the direct problem (the operators not being supported), but the use of an ES6 `Map` and `Object.entries()` in [colors.js](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/colors.js#L228) still causes problems, because some of those features are only implemented in [JDK 9](https://openjdk.java.net/jeps/292).

I still have to figure out how to get Babel to transpile those to ES5.

The fix is being implemented in #10737.
